fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/leaderboard/LeaderboardIndexTest.java

```diff
@@ -527,6 +530,52 @@ public void basicUngroupedNestedReverse() {
         }
     }
 
+    @ParameterizedTest(name = "emptyUngroupedNested[highScoreFirst={0}]")
+    @BooleanSource
+    public void emptyUngroupedNested(boolean highScoreFirst) {
+        Leaderboards leaderboards = new UngroupedNestedLeaderboards();
+        basicSetup(leaderboards, highScoreFirst);
+        try (FDBRecordContext context = openContext()) {
+            leaderboards.openRecordStore(context, false);
+
+            // Save a record for evaluating record functions later
+            final FDBStoredRecord<Message> rec = leaderboards.findByName("helen");
+            leaderboards.recordStore.deleteAllRecords();
+
+            assertEquals(Collections.emptyList(),
+                    leaderboards.scanIndexByRank(TupleRange.ALL).asList().join());
+
+            TupleRange top3 = new TupleRange(Tuple.from(0), Tuple.from(2), EndpointType.RANGE_INCLUSIVE, EndpointType.RANGE_INCLUSIVE);
+            assertEquals(Collections.emptyList(),
+                    leaderboards.scanIndexByRank(top3).asList().join());
+
+            IndexAggregateFunction allTimeFunction = leaderboards.scoreForTimeWindowRank(TimeWindowLeaderboard.ALL_TIME_LEADERBOARD_TYPE, -1);
+            assertNull(leaderboards.evaluateAggregateFunction(allTimeFunction, Tuple.from(42)));
+
+            assertNull(leaderboards.evaluateQueryFunction(leaderboards.queryRank(), rec));
+            assertNull(leaderboards.evaluateQueryFunction(leaderboards.queryTimeWindowRank(TimeWindowLeaderboard.ALL_TIME_LEADERBOARD_TYPE, -1), rec));
+            assertNull(leaderboards.evaluateQueryFunction(leaderboards.queryTimeWindowRank(TEN_UNITS, 10102), rec));
+        }
+    }
+
+    @ParameterizedTest(name = "missingFromUngrouped[highScoreFirst={0}]")
+    @BooleanSource
+    public void missingFromUngrouped(boolean highScoreFirst) {
```

Comment:
Both this test and the one below would fail with an NPE prior to the fix in this PR